### PR TITLE
Fixing unicode issue when used in shell environment

### LINKED ISSUE #314: Unicode in environment variables break ansible-runner
We run ansible-runner in Docker containers in Gitlab CI.
ansible-runner breaks when a user with Umlauts in his name starts the runner, as the user's name (eg. `Müller`) is saved as environment variable.

```console
[root@af4df578c2e7 /]# unset test
[root@af4df578c2e7 /]# ansible-runner -p test.yml run .
ERROR! the playbook: test.yml could not be found
[root@af4df578c2e7 /]# export test=Müller
[root@af4df578c2e7 /]# ansible-runner -p test.yml run .
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/ansible_runner/__main__.py", line 329, in main
    res = run(**run_options)
  File "/usr/lib/python2.7/site-packages/ansible_runner/interface.py", line 162, in run
    r.run()
  File "/usr/lib/python2.7/site-packages/ansible_runner/runner.py", line 112, in run
    'env': self.config.env}, f, ensure_ascii=False
  File "/usr/lib64/python2.7/json/__init__.py", line 190, in dump
    fp.write(chunk)
  File "/usr/lib64/python2.7/codecs.py", line 691, in write
    return self.writer.write(data)
  File "/usr/lib64/python2.7/codecs.py", line 351, in write
    data, consumed = self.encode(object, self.errors)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 2: ordinal not in range(128)
[root@af4df578c2e7 /]# ansible-runner --version
1.3.4
```

## related issues

similar to #234

## link to breaking source

https://github.com/ansible/ansible-runner/blob/1696ec2075f96b57bc9099cf018ed8d0d574aed6/ansible_runner/runner.py#L109-L113

### PR DESCRIPTION
**Pull request by matburt**
Fixes #314

### CODE REVIEW

**Comment by matburt** [comment]
recheck

**Comment by matburt** [comment]
recheck

**Comment by matburt** [comment]
recheck

File: test/integration/test_main.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
+# -*- coding: utf-8 -*-
```
Comment (benthomasson) [inline test/integration/test_main.py]: Coding with emojis ftw!

**Pull request merged**